### ISSUE TITLE
How to debug using qtdeploy

### ISSUE THREAD

**Issue by ismlsmile** [therecipe/qt#600]
I want debug the application using "qtdeploy test xxx", how could I do it ?

**Comment by amlwwalker** [comment]
Run the App directly. When u run qtdeploy test desktop it opens the app with a hidden console. If you run the binary (found for osx in) deploy/app name.app/Contents/.../macos/appname, from the terminal then u get your debug logs

**Comment by ismlsmile** [comment]
Thanks for your reply.
I mean I want set some breakpoints, could I do that?

**Comment by therecipe** [comment]
@ismlsmile I had some luck with https://github.com/derekparker/delve some time ago.
But there was some issue with Go >= 1.8 (can't recall what exactly) that did broke the support for cgo programs or something. I haven't looked into delve in a long time, so maybe they have fixed this by now. I also did test gdb on linux and it did work iirc.

**Comment by ismlsmile** [comment]
I got it, thanks.
By the way, how o you debug when developing application based on this library. using gdb or just println?

**Comment by creedr** [comment]
Hello, I am trying to debug an app with println on my windows machine. From the discussion above, it seems like running `qtdeploy test desktop` then running the binary from the console should keep the command running in the console until the app window is closed (i.e. the command prompt won't show up until the app is closed) and thus allow output from `println` to print on the console. Is this right? I cant seem to get any console command to stay 'open' while my app is running. I tried commenting out the ldFlags `"-H=windowsgui"` in cmd.go and re-running qtsetup, but this is still not doing what I am expecting.

**Comment by arsham** [comment]
@creedr you can use `go run ...` to see your print statements. Just remember to invoke `qtrcc` and `qtmoc` once (however, qtdeploy does that for you).

**Comment by creedr** [comment]
@arsham Hmm, thats not working either. `qtrcc` and `qtmoc` have been inviked already. Does `qtdeploy run` invoke the `go run ...` command? `qtdeploy run` launches the app, but still no console with print statements.

**Comment by arsham** [comment]
Can you leave us an example application please?

**Comment by arsham** [comment]
You probably won't see the print outputs if you use `qtdeploy`. However you should be able to see them by using the `go` command. Your problem is the security feature in `cgo` hence the `invalid flag in #cgo..` message. You can work around that by exporting these variables in the shell: 
```bash
export CGO_CXXFLAGS_ALLOW=".*"
export CGO_LDFLAGS_ALLOW=".*"
export CGO_CFLAGS_ALLOW=".*"
```
This works on Linux, but I'm not sure how you'd do it on windows. Hope that helps.

**Comment by creedr** [comment]
Thanks for the help @arsham  I finally managed to get the Println outputs on the console, but not in the way I was originally trying to do.  I ended up installing a new virtual machine with Ubuntu, then when I finally got Go and Qt stuff installed, the Println statements appeared only when I ran `go build` and started the app from the console. I guess for now I will be developing my Go/Qt apps on my VM, then deploy on Windows.
